Fluidity_Optimize_Parallel_Example.py

```diff
@@ -204,27 +204,31 @@ def add_string(a):
     # loop through each processor ID used to generate Fluidity model.
     for i in range(1,29,1):
```

Comment:
Why `29`?

Reply:
Normally, we run the geodynamic model on 28 processors, so particles can have processor IDs between 1 and 28. In the example I have attached it is between 1 and 8.

Reply:
You store information about your MPI initialization here.
https://github.com/paddywball/BDD21/blob/241b303a5af59f9debbe1c5bc13345d8f9a3cdeb/Fluidity_Optimize_Parallel_Example.py#L53-L55

You could use `max_cpu` here instead of the "magic number" `29`.